app/controllers/home_controller.rb

```diff
@@ -191,19 +191,23 @@ def saved
   end
 
   def tagged
-    @tag = Tag.where(:tag => params[:tag]).first!
+    tag_params = params[:tag].split(',')
+    @tags = Tag.where(tag: tag_params)
+    @tags.first!
```

Comment:
```suggestion
    raise ActiveRecord::RecordNotFound unless @tags.length == tag_params.length
```

If any tag asked for isn't found, this should 404. Not just to indicate the error instead of silently doing the wrong thing, but because the `params[:tag]` can be printed back to the user.